tracks/julia/exercises/pythagorean-triplet/mentoring.md

```diff
@@ -25,34 +55,51 @@ function pythagorean_triplets(n)
     return triplets
 end
 ```
-````
 
-Telling Julia the type of `triplets` allows it to store the values without indirection. Without an annotation some (all?) version of Julia will type `triplets` as `Vector{Any}`, which means a vector of pointers to heap-allocated objects, which is slower.
+Calculating c from a and b clearly eliminates a huge number of iterations and is the main reason for the speedup.
 
-A small improvement is to tighten up the loop bounds using some simple arithmetic and ceiling division (`cld`):
+A much more modest improvement is achieved by telling Julia the type of `triplets` so that it can store the values without indirection.
+Without an annotation some (all?) version of Julia will type `triplets` as `Vector{Any}`, which means a vector of pointers to heap-allocated objects, which is slower.
+Annotating with an immutable type like we have done (which by definition will be of known size) allows Julia to simply store the items one after the other without indirection.
 
-```
+We can improve this solution slightly by tightening up the loop bounds using some simple arithmetic and floor division (`fld`):
+
+```julia
     # Lower bound because the smallest triple is 3, 4, 5.
     # Upper bounds implied by a < b < c && a + b + c == n.
-    for a in 3:cld(n, 3)
-        for b in a+1:cld(2n, 3)
+    for a in 3:fld(n, 3)-1
+        for b in a+1:fld(n, 2)-1
 ```
 
-This linear time algorithm derived by solving some simultaneous equations is a much greater improvement:
+> Note: you can come up with even tighter bounds than these. Share your working with your mentor if you do!
+
+The intuition on upper bounds is that if we were working with real numbers, the biggest `a` can be is `n/3 - ε`,
+where `ε` is the smallest real number,
```

Comment:
This is a bit misleading, I think. The smallest real number is 0 (unless we're considering negatives). Maybe

```suggestion
where `ε` is the smallest non-zero real number,
```

if you want to avoid the usual "Let ε > 0, ..."?

Reply:
Good catch. I'll use:

```suggestion
where `ε` is the smallest real number greater than 0,
```